examples/cli/src/main/java/io/unitycatalog/cli/utils/CliUtils.java

```diff
@@ -151,14 +147,145 @@ public static String preprocess(String value, int length) {
+        List<String> fields = getFields(node.get(0));
+
+        int[] columnWidths = new int[fields.size()];
+        boolean[] fixedWidthColumns = new boolean[fields.size()];
+        int maxLineWidth = calculateFieldWidths(node, fields, columnWidths, fixedWidthColumns);
+
+        if (maxLineWidth >= outputWidth) {
+            adjustColumnWidths(fields, outputWidth, columnWidths, fixedWidthColumns);
+        }
+
+        List<String> headers = fields.stream().map(String::toUpperCase).collect(Collectors.toList());
+
+        CWC_FixedWidth cwc = new CWC_FixedWidth();
+        for (int i = 0; i < fields.size(); i++) {
+            cwc.add(columnWidths[i]);
+        }
+
+        at.getRenderer().setCWC(cwc);
+        at.addRule();
+        at.addRow(headers.toArray()).setTextAlignment(TextAlignment.CENTER);
+        at.addRule();
+        node.forEach(element -> {
+            List<String> row = new ArrayList<>();
+            ObjectNode objectNode = (ObjectNode) element;
+            Iterator<String> fieldNames = objectNode.fieldNames();
+            int columnIndex = 0;
+            while (fieldNames.hasNext()) {
+                String fieldName = fieldNames.next();
+                JsonNode value = element.get(fieldName);
+                String valueString = value.isTextual() ? value.asText() : value.toString();
+                row.add(preprocess(valueString, columnWidths[columnIndex]));
+                columnIndex++;
+            }
+            at.addRow(row.toArray());
+            at.addRule();
+        });
+    }
+
+    private static void processOutputAsKeysAndValues(AsciiTable at, JsonNode node, int outputWidth) {
+        int minOutputWidth = outputWidth / 2;
+
+        at.getRenderer().setCWC(new CWC_LongestLine()
+                .add(minOutputWidth / 3,outputWidth / 4)
+                .add(2 * minOutputWidth / 3, 3 * outputWidth / 4));
+        at.addRule();
+        at.addRow("KEY", "VALUE").setTextAlignment(TextAlignment.CENTER);
+        at.addRule();
+        node.fields().forEachRemaining(field -> {
+            JsonNode value = field.getValue();
+            StringBuilder valueString = new StringBuilder();
+            if (value.isTextual()) {
+                valueString = new StringBuilder(value.asText());
+            } else if (value.isArray()) {
+                ArrayNode arrayNode = (ArrayNode) value;
+                for (int i = 0; i < arrayNode.size(); i++) {
+                    valueString.append(arrayNode.get(i).toString()).append("\n\n");
+                }
+            } else {
+                valueString = new StringBuilder(value.toString());
+            }
+            at.addRow(field.getKey().toUpperCase(), valueString.toString()).setTextAlignment(TextAlignment.LEFT);
+            at.addRule();
+        });
+
+    }
+
+    private static int getOutputWidth() {
+        // TODO: maybe turn this into a command line parameter.
+        String envWidth = System.getenv().get("UC_OUTPUT_WIDTH");
```

Comment:
This is specifically to handle the failure in tutorial tests?

Reply:
Yes. I didn't look into whether you could set a "terminal width" in github actions, but I doubt it. Alternatively I could change the logic to be... if we get a non-zero width from the terminal then use it, otherwise use the original 180 character width.

Reply:
I think it is fine for now, given we have a fallback default